utils/db.py

```diff
@@ -92,12 +93,14 @@ class Db(Mapping):
         tables, like the mapping interface or :py:meth:`values`.
 
     """
-    def __init__(self, hostname=None, credentials=None):
+    def __init__(self, hostname=None, credentials=None, db_port=None):
         self._table_cache = {}
-        if hostname is None:
+        if hostname is None and db_port is None:
```

Comment:
@quarckster So, problem solved then, you can safely do `return Db(appliance)` and Db can count on appliance.db_address :)